src/components/network/network.js

```diff
@@ -109,27 +109,40 @@ export default class Network {
         );
     }
 
-    updateEquipments(currentEquipments, newEquipements) {
+    updateEquipments(currentEquipments, newEquipements, equipmentType) {
         // replace current modified equipments
         currentEquipments.forEach((equipment1, index) => {
             const found = newEquipements.filter(
                 (equipment2) => equipment2.id === equipment1.id
             );
             currentEquipments[index] = found.length > 0 ? found[0] : equipment1;
         });
+
         // add newly created equipments
-        newEquipements.forEach((equipment1) => {
-            const found = currentEquipments.find(
-                (equipment2) => equipment2.id === equipment1.id
-            );
-            if (found === undefined) {
-                currentEquipments.push(equipment1);
-            }
-        });
+        let equipmentsAdded = false;
+        if (this.isResourceFetched(equipmentType)) {
```

Comment:
if `isResourceFetched` use an `equipmentType` then we should rename its parameter which is `equipment` and which is not really undestandable. For homogenous code, we could rename equipmentType to equipment here but I prefer the first solution. What do you think of this ?